src/pages/index.js

```diff
@@ -45,30 +73,68 @@ const section = new Section(
 );
 section.renderItems();
 
-const userInfo = new UserInfo({
-  nameSelector: ".profile__title",
-  descriptionSelector: ".profile__description",
+// retrieve use info
+api.getUserInfo().then((data) => {
+  // console.log(data);
+  userInfo.setUserInfo({
+    name: data.name,
+    about: data.about,
+    avatar: data.avatar,
+  });
 });
 
+function handleProfileFormSubmit(inputValues) {
+  userInfo.setUserInfo({
+    name: inputValues.name,
+    about: inputValues.about,
+    avatar: inputValues.avatar,
+  });
+
+  return api
+    .setUserInfo({
+      name: inputValues.name,
+      about: inputValues.about,
+      avatar: inputValues.avatar,
+    })
+    .then((data) => {
+      // after updating, reflect the changes in the UI
+      userInfo.setUserInfo({
+        name: data.name,
+        about: data.about,
+        avatar: data.avatar,
+      });
+      profilePopup.close();
+    });
+}
+
+/*=============================================
+=               Handle avatar                 =
+=============================================*/
+// instantiate the avatar popup
+const avatarPopup = new PopupWithForm(
+  "#avatar-picture-modal",
+  handleAvatarFormSubmit
+);
+avatarPopup.setEventListeners();
+
+function handleAvatarFormSubmit(inputValues) {
+  // pass avatar url to the API
+  return api.setUserAvatar(inputValues.avatar).then((updatedData) => {
+    userInfo.setUserInfo(updatedData); // update avatar in teh DOM
+    avatarPopup.close();
+  });
+}
+
 /*=============================================
 =             Form Validation                 =
 =============================================*/
 const editFormValidation = new FormValidator(settings, profileEditForm);
 const addFormValidation = new FormValidator(settings, newCardForm);
+const avatarFormValidation = new FormValidator(settings, avatarForm);
```

Comment:
**COULD BE IMPROVED**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```